File: output/page_file_generator/Equfix_home_page_selenium.java
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.How;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.openqa.selenium.support.ui.ExpectedConditions;
import java.time.Duration;
import java.util.List;
import java.util.ArrayList;

public class Equfix_home_page {

    private WebDriver driver;
    private WebDriverWait wait;

    @FindBy(how = How.XPATH, using = "//a[contains(@class, 'btn') and text()='PLACE AN ALERT']")
    private WebElement placeAlertButton;

    @FindBy(how = How.XPATH, using = "//button[@name='close']")
    private WebElement closeButton;

    public Equfix_home_page(WebDriver driver) {
        this.driver = driver;
        this.wait = new WebDriverWait(driver, Duration.ofSeconds(20));
        PageFactory.initElements(driver, this);
    }

    private void waitForElement(WebElement element) {
        wait.until(ExpectedConditions.visibilityOf(element));
    }

    private void waitForElementToBeClickable(WebElement element) {
        wait.until(ExpectedConditions.elementToBeClickable(element));
    }

    public void switchToNewWindow() {
        List<String> handles = new ArrayList<>(driver.getWindowHandles());
        if (handles.size() > 1) {
            driver.switchTo().window(handles.get(handles.size() - 1));
        }
    }

    public void clickPlaceAlertButton() {
        waitForElementToBeClickable(placeAlertButton);
        placeAlertButton.click();
    }

    public void clickCloseButton() {
        waitForElementToBeClickable(closeButton);
        closeButton.click();
    }
}

File: output/page_file_generator/Equfix_home_page_selenium.py
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC

class Equfix_home_page_selenium:
    LOCATORS = {
        "place_an_alert_btn": (By.XPATH, "//a[contains(@class, 'btn') and text()='PLACE AN ALERT']"),
        "close_btn": (By.XPATH, "//button[@name='close']")
    }

    def __init__(self, driver):
        self.driver = driver
        self.wait = WebDriverWait(driver, 20)

    def wait_for_element(self, locator_key):
        if locator_key not in self.LOCATORS:
            raise RuntimeError(f"Locator for '{locator_key}' not found.")
        return self.wait.until(EC.presence_of_element_located(self.LOCATORS[locator_key]))

    def click_element(self, locator_key):
        element = self.wait_for_element(locator_key)
        element.click()

    def enter_text(self, locator_key, text):
        element = self.wait_for_element(locator_key)
        element.clear()
        element.send_keys(text)

    def switch_to_new_window(self):
        if len(self.driver.window_handles) > 1:
            self.driver.switch_to.window(self.driver.window_handles[-1])

    def place_an_alert(self):
        self.switch_to_new_window()
        self.click_element("place_an_alert_btn")
        alert_btn = self.wait_for_element("place_an_alert_btn")
        self.enter_text("place_an_alert_btn", "PLACE AN ALERT")